tracing.go

```diff
@@ -484,42 +488,42 @@ func (s *Span) sample() Sampled {
 	if sampler != nil {
 		tracesSamplerSampleRate := sampler.Sample(samplingContext)
 		s.sampleRate = tracesSamplerSampleRate
-		if tracesSamplerSampleRate < 0.0 || tracesSamplerSampleRate > 1.0 {
-			Logger.Printf("Dropping transaction: Returned TracesSampler rate is out of range [0.0, 1.0]: %f", tracesSamplerSampleRate)
-			return SampledFalse
-		}
-		if tracesSamplerSampleRate == 0 {
-			Logger.Printf("Dropping transaction: Returned TracesSampler rate is: %f", tracesSamplerSampleRate)
-			return SampledFalse
+		// tracesSampler can update the sample_rate on frozen DSC
+		if len(s.dynamicSamplingContext.Entries) > 0 {
+			s.dynamicSamplingContext.Entries["sample_rate"] = strconv.FormatFloat(tracesSamplerSampleRate, 'f', -1, 64)
 		}
-
-		if rng.Float64() < tracesSamplerSampleRate {
-			return SampledTrue
-		}
-		Logger.Printf("Dropping transaction: TracesSampler returned rate: %f", tracesSamplerSampleRate)
-		return SampledFalse
+		return samplingDecision(s.sampleRate, "TracesSampler")
 	}
+
 	// #4 inherit parent decision.
-	if s.parent != nil {
-		Logger.Printf("Using sampling decision from parent: %v", s.parent.Sampled)
-		switch s.parent.Sampled {
+	if s.Sampled != SampledUndefined {
+		Logger.Printf("Using sampling decision from parent: %v", s.Sampled)
+		switch s.Sampled {
```

Comment:
Maybe let's expose an additional `parentSampeled` property on the span, so we don't have any side effects of people explicitly setting `Sampled` on a span.

Reply:
Not required